### PR TITLE
[EG] remove preview versions

### PR DESCRIPTION
**Pull request by l0lawrence**
TSP:
- Removing preview ApiVersions from Event Grid Namespaces typespec.

Readme:

- Reformatting Readme.md to remove Preview Api Versions
- Reorder api versions from most recent to least recent (alt pr #31815)
- Removing system events from namespaces api_versions as they are not a part of the service contract. (alt pr #31815)

### CODE REVIEW

File: specification/eventgrid/data-plane/readme.md [hunk 1]
```diff
@@ -111,118 +111,35 @@ openapi-type: data-plane
...
-### Tag: package-2018-01
+### Tag: package-2024-06-01
 
-These settings apply only when `--tag=package-2018-01` is specified on the command line.
+These settings apply only when `--tag=package-2024-06-01` is specified on the command line.
 
-``` yaml $(tag) == 'package-2018-01'
+``` yaml $(tag) == 'package-2024-06-01'
 input-file:
-- Microsoft.Storage/stable/2018-01-01/Storage.json
```
Comment (l0lawrence) [inline specification/eventgrid/data-plane/readme.md:120]: Reformatting order of api - versions from recent to least recent and removing system events from namespace versions

File: specification/eventgrid/Azure.Messaging.EventGrid/main.tsp [hunk 2]
```diff
@@ -340,8 +332,6 @@ namespace Microsoft.EventGrid {
 
   // RenewLock Operation:
   // POST https://{namespaceName}.{region}.eventgrid.azure.net/topics/{topicName}/eventsubscriptions/{eventSubscriptionName}:renewLock?api-version={apiVersion}
-  @added(ServiceApiVersions.v2023_10_01_preview)
-  @removed(ServiceApiVersions.v2023_11_01)
```
Comment (l0lawrence) [inline specification/eventgrid/Azure.Messaging.EventGrid/main.tsp:344]: can remove this since we have the @added below

Comment (lmazuel) [reply specification/eventgrid/Azure.Messaging.EventGrid/main.tsp:344]: I'm glad, it was horrible to have this added/removed/added

**Review by lmazuel — APPROVED** [review]
Discussed with @JeffreyRichter that those previews need to be deleted

File: specification/eventgrid/data-plane/readme.md (hunk 1 above)
Comment (l0lawrence) [reply specification/eventgrid/data-plane/readme.md]: will be done in separate pr

**Pull request merged**